Xamarin.Forms.Platform.iOS/VisualElementPackager.cs

```diff
@@ -141,34 +143,51 @@ protected virtual void OnChildRemoved(VisualElement view)
 			viewRenderer.Dispose();
 		}
 
-		void EnsureChildrenOrder()
+		void EnsureChildrenOrder(VisualElement element = null)
 		{
 			if (ElementController.LogicalChildren.Count == 0)
 				return;
 
-			for (var z = 0; z < ElementController.LogicalChildren.Count; z++)
+			var effectedChildIndex = 0;
+			if (element != null)
+				effectedChildIndex = ElementController.LogicalChildren.IndexOf(element);
+
+			for (var z = effectedChildIndex; z < ElementController.LogicalChildren.Count; z++)
 			{
 				var child = ElementController.LogicalChildren[z] as VisualElement;
 				if (child == null)
 					continue;
-				var childRenderer = Platform.GetRenderer(child);
+				OrderElement(child, z);
+			}
+		}
 
-				if (childRenderer == null)
-					continue;
+		void OrderElement(VisualElement element, int order)
+		{
+			Performance.Start(out string reference);
+			if (CompressedLayout.GetIsHeadless(element))
+				return;
+
+			var childRenderer = Platform.GetRenderer(element);
+
+			if (childRenderer == null)
+				return;
 
-				var nativeControl = childRenderer.NativeView;
+			var nativeControl = childRenderer.NativeView;
 #if __MOBILE__
-				Renderer.NativeView.BringSubviewToFront(nativeControl);
+			Renderer.NativeView.BringSubviewToFront(nativeControl);
 #endif
-				nativeControl.Layer.ZPosition = z * 1000;
-			}
+			nativeControl.Layer.ZPosition = order * 1000;
+			Performance.Stop(reference);
```

Comment:
this will never be called if we return on line 173

Reply:
And also if we return on line 168. @rmarinho can you fix this ? It looks like @shanranm is inactive on github.

Reply:
Fixed these!